submissions/Roophee/a-tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,65 @@
+"use strict";
+
+class Organism {
+    constructor(name, gender, saying, legs, hands, tail) {
+        this.name = name;
+        this.gender = gender;
+        this.species = this.constructor.name.toLowerCase();
+        this.saying = saying;
+        this.legs = legs;
+        this.hands = hands;
+        this.tail = tail;
+        this.friends = [];
+    }
+
+    addFriends(...arrayOfFriends) {
+        arrayOfFriends.forEach(({ name }) => this.friends.push(name));
+    }
+
+    description() {
+        let descriptionArray = [this.name, this.gender, this.species, this.saying, this.legs, this.hands, this.tail];
+        this.friends.length === 0 ? descriptionArray.push('none') : descriptionArray.push(this.friends);
```

Comment:
```suggestion
        let descriptionArray = [this.name, 
          this.gender,
          this.species,
          this.saying,
          this.legs,
          this.hands,
          this.tail,
          this.friends.length === 0 ? 'none' : this.friends,
        ];
```

But beware: a class cannot handle properties it doesn't own or inherit from ancestors.
In good OOP design a class handles only props it owns.
Method overloading is to the rescue.

Reply:
I rewrote code and add to each property a label with name of property, because it will be difficult to understand what mean property value on method overloading and length of returned string will be different depending on class. I think, it will be more recognizably.